scale-tests/kafka_cassandra_streaming_test.py

```diff
@@ -175,58 +202,70 @@ def append_submission(output_file: str, dispatcher_app_name: str, submission_id:
     cassandra_num_nodes           = cassandra['nodes']['count']
-    num_producers_per_dispatcher  = int(args['--num-producers-per-dispatcher'])
+    num_producers_per_kafka       = int(args['--num-producers-per-kafka'])
     num_consumers_per_producer    = int(args['--num-consumers-per-producer'])
+    producer_must_fail            = args['--producer-must-fail']
     producer_number_of_words      = int(args['--producer-number-of-words'])
     producer_words_per_second     = int(args['--producer-words-per-second'])
     producer_spark_cores_max      = int(args['--producer-spark-cores-max'])
     producer_spark_executor_cores = int(args['--producer-spark-executor-cores'])
+    consumer_must_fail            = args['--consumer-must-fail']
     consumer_write_to_cassandra   = args['--consumer-write-to-cassandra']
     consumer_batch_size_seconds   = int(args['--consumer-batch-size-seconds'])
     consumer_spark_cores_max      = int(args['--consumer-spark-cores-max'])
     consumer_spark_executor_cores = int(args['--consumer-spark-executor-cores'])
 
-    producer_must_fail = args['--producer-must-fail']
-    consumer_must_fail = args['--consumer-must-fail']
-
-    log.info("Dispatchers: \n{}".format("\n".join(dispatchers)))
+    num_kafkas = len(kafkas)
+    num_dispatchers = len(dispatchers)
+    num_producers = num_kafkas * num_producers_per_kafka
+    num_consumers = num_producers * num_consumers_per_producer
+    num_jobs = num_producers + num_consumers
+    avg_num_jobs_per_dispatcher = num_jobs / num_dispatchers
+    max_num_jobs_per_dispatcher = math.ceil(avg_num_jobs_per_dispatcher)
+
+    log.info('Number of Kafka clusters: {}'.format(num_kafkas))
+    log.info('Number of dispatchers: {}'.format(num_dispatchers))
+    log.info('Total number of jobs: {} ({} producers, {} consumers)'.format(
+        num_jobs, num_producers, num_consumers))
+    log.info('Average number jobs per dispatcher: {}'.format(avg_num_jobs_per_dispatcher))
+    log.info('Will run at most {} jobs per dispatcher'.format(max_num_jobs_per_dispatcher))
+    log.info("\nDispatchers: \n{}\n".format("\n".join(dispatchers)))
+
+    dispatchers_slots = sdk_utils.mapcat(make_repeater(max_num_jobs_per_dispatcher),
+                                         dispatchers)
 
     _install_package_cli(kafka_package_name)
     _install_package_cli(cassandra_package_name)
     _install_package_cli(SPARK_PACKAGE_NAME)
 
-    kafka_broker_dns = _service_endpoint_dns(kafka_package_name, kafka_service_name, "broker")
     cassandra_native_client_dns = _service_endpoint_dns(cassandra_package_name, cassandra_service_name, "native-client")
 
-    for dispatcher in dispatchers:
-        dispatcher_app_name, _, driver_role = dispatcher.split(",")
+    for kafka in kafkas:
+        kafka_num_brokers = kafka['brokers']['count']
+        kafka_service_name = kafka['service']['name']
+        kafka_broker_dns = _service_endpoint_dns(kafka_package_name, kafka_service_name, 'broker')
```

Comment:
The `kafka_package_name` could also be made local to do away with the assumption that we have uniform package names.

Reply:
https://github.com/mesosphere/spark-build/pull/336#discussion_r191400753

Reply:
cca684cc7011a7210374d3b3cc6085e315be4ad1